go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/iotexproject/iotex-core/v2
 
-go 1.21.11
+go 1.22.12
```

Comment:
We may need further checks for go1.22 upgrades. I have created a issue #4560 to track related things.